droidlet/tools/hitl/nsp_retrain_infra.py

```diff
@@ -0,0 +1,238 @@
+        f.write("valid acc " + str(macc)  + "\n")
+        f.close()
+
+    def downloadData(self):
+        base_data_dir = os.path.join(opts.droidlet_dir, opts.full_data_dir)
+        batch_id = str(self.batch_id)
+        download_dir = os.path.join(base_data_dir, batch_id) # Currently downloads data to a new dir each time
+        try:
+            os.mkdir(download_dir)
+        except FileExistsError:
+            pass
+        data_filepath = download_dir + '/nsp_data.txt'
+        try:
+            s3.download_file('droidlet-hitl', 'nsp_data.txt', data_filepath)  # Will overwrite file if exists
+        except:
+            logging.info(f"Exception raised on S3 file download")
+            raise
+        logging.info(f"New data download completed successfully")
+
+        # Check the formatting of the file, strip off the index and resave
+        annotated_filepath = download_dir + '/annotated.txt'  # Change name to be what sweep_runner expects
+        with open(data_filepath, "r") as nspfile, open(annotated_filepath, "w") as afile:
+            for line in nspfile:
+                parsed = line.split('|')
+                if len(parsed) == 2:
+                    afile.write(line)
+                elif len(parsed) == 3:  # Cut off the index if it exists
+                    afile.write(str(parsed[1]) + '|' + str(parsed[2]))
+                else:
+                    raise ValueError("Annotated NSP data & logical forms not formatted as expected")
+        logging.info(f"data successfully preprocessed into annotated.txt")
+
+        return batch_id, download_dir
+
+
+    def run(self):
+        logging.info(f"NSP Retraining Job initialized, downloading new data")
+        batch_id, download_dir = self.downloadData()
+    
+        # Setup sweep_runner args
+        full_data_dir = download_dir[len(opts.droidlet_dir):] + "/"  # Need to slice off the base droidlet filepath b/c sweep_runner adds it back
+        sweep_args = "python3 " + \
+            os.path.join(opts.sweep_runner_dir, "sweep_runner.py") + \
+            " --sweep_config_folder " + opts.sweep_config_folder + \
+            " --sweep_scripts_output_dir " + opts.sweep_scripts_output_dir + \
+            " --checkpoint_dir " + opts.checkpoint_dir + \
+            " --sweep_name " + batch_id + \
+            " --output_dir " + opts.output_dir + \
+            " --droidlet_dir " + opts.droidlet_dir + \
+            " --full_data_dir " + full_data_dir
+
+        # Initialize the training run
+        try:
+            logging.info(f"Executing in new shell: \n{sweep_args}")
+            sweep = Popen(sweep_args, shell=True, stdout=PIPE, stderr=PIPE, text=True)
+        except OSError:
+            logging.info(f"Likely error: sweep_runner.py not found where expected")
+            raise
+        except ValueError:
+            logging.info(f"Likely error: Popen called with invalid arguments")
+            raise
+
+        # TODO replace with while loop and poll
```

Comment:
let's prioritize this TODO because it will help with debugging when integrated into the full pipeline